feat(docs): compilation report

novusnota opened the pull request:
<!--
IMPORTANT:
If your PR doesn't close a particular issue, please, create the issue first and describe the whole context: what you're adding/changing and why you're doing so. And only then open the Pull Request, which would close that issue!
-->

Also adjusted the generated report itself, just so it's a valid and readable markdown.

## Issue

Closes #1094.
Towards #1136 (created the page).

## Checklist

- [x] I have updated CHANGELOG.md
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

novusnota committed:
feat(docs): compilation report

anton-trunov commented on docs/src/content/docs/book/compile.mdx:
```diff
@@ -0,0 +1,104 @@
+
+:::note
+
+  This page is still being written as per [#1136](https://github.com/tact-lang/tact-docs/issues/1136).
```
the link gives 404

novusnota replied:
Fixed, nice catch

anton-trunov commented on docs/src/content/docs/book/compile.mdx:
```diff
@@ -0,0 +1,104 @@
+
+The first section introduces the present structures, i.e. some of the [composite types](/book/types#composite-types), [Structs and Messages](/book/structs-and-messages) that are declared or imported directly in the contract code, as well as those exposed from the Core standard library.
+
+Along with the number of structures present, each of the [Structs][struct] and [Messages][message] are described with their respective [TL-B schemas][tlb] and Tact signatures.
```
Let's mention somewhere (with an example) that you can find auto-generated message opcodes in the compilation report

novusnota replied:
Done

novusnota committed:
fix: suggestions from the code review

novusnota committed:
whoopsie

novusnota committed:
not a Tact signature, strictly speaking

Commit:
Merge branch 'main' into closes-1094-comp-rep

anton-trunov commented on docs/src/content/docs/book/compile.mdx:
````diff
@@ -0,0 +1,116 @@
+### Seven
+TL-B: `seven#00000007 = Seven`
+Signature: `Seven{}`
+```
````
Let's
- make it a subsection with its own link
- link the subsection from the messages page
- be more explicit, i.e. provide the corresponding Tact message declarations to their respective TL-B schemes
- explicitly say `0x946a98b6` and `0x00000007` are the opcodes
- mention that the opcodes are in hex as it is important for something like `foobar#94609816`

novusnota replied:
Max heading level is intentionally set to 4, so anything beyond (`#####` or `######`) won't create linked headings. Therefore, I've linked the whole subsection and rewrote its contents.

anton-trunov commented on docs/src/content/docs/book/compile.mdx:
```diff
@@ -0,0 +1,116 @@
+
+There, the `lshift(){:tact}` getter has a single argument `x`, whereas the `gas(){:tact}` getter has no arguments.
+
+#### Exit codes {#exit-codes}
```
we should link this subsection from the exit codes page

novusnota replied:
Done

novusnota committed:
Merge branch 'main' into closes-1094-comp-rep

novusnota committed:
fix: suggestions from the code review

novusnota committed:
fix: correction of the `queryId` description

This didn't warrant a separate issue, but was important to do as pointed
out in Tact chat

novusnota commented:
Also corrected the description of `queryId` in Jettons to match the DeDust & StonFi cookbook pages, as [pointed out in the Tact chat](https://t.me/tactlang/51785) by @Shvandre. I felt like it didn't warrant a separate issue, but was important to address somewhere, so here we are :)

novusnota committed:
Merge branch 'main' into closes-1094-comp-rep

novusnota committed:
Merge branch 'main' into closes-1094-comp-rep

anton-trunov reviewed (approved):
the compilation report LGTM, waiting for a review of the jetton-related part by @Shvandre

Shvandre reviewed (approved):
🚀
